docs/content/any/getting-started/quickstart/index.md

```diff
@@ -1,192 +1,246 @@
+
+A _resource_ rule governs access to a specific resource in your app — for instance, a page, an object, a route, or a database entry.
+Here, there are two possible actions that can be done to a Page: “read” and “write.”
+Users can read, but not write, to a page.
+Admins are allowed to write to a Page.
+The line `implies: [“user”]` says that admins can also do anything Users are allowed to do; that is, read a Page.
+
+There’s much more you can do with Polar, including defining parent-child relationships — we’re just scratching the surface here.
+
+## Calling back into your Python code
+
+You can call properties and methods on your Python objects from Polar.
+These will defer control back to your app.
+Oso leaves the decision of how to store role assignments up to you — you might choose to store those role assignments in a database, in memory, or create them dynamically.
+Our `actor_role` rule calls your Python method `has_roles` to get all the roles for our actor.
+
+```polar
+actor_role(actor, role) if
+     resources = Page.pages and
+     r in resources and
+     actions = r.has_roles(actor) and
+     action in actions and
+     role = { name: action, resource: r };
+ ```
+
+To access properties on your Python objects, you must `register` their classes with Oso.
+
+```python
+from oso import Oso
+oso = Oso()
+
+oso.register_class(Page)
+oso.register_class(User)
 ```
 
-  ```console
-  $ curl -H "user: alice@example.com" localhost:5050/expenses/3
-  Not Authorized!
-  ```
-{{% /callout %}}
-
-Feel free to play around with the current policy and experiment with adding
-your own rules!
-
-For example, if you have `Expense` and `User` classes defined in your
-application, you could write a policy rule in Oso that says a `User` may
-`"approve"` an `Expense` if they manage the `User` who submitted the expense
-and the expense’s amount is less than $100.00:
-
-{{< code file="expenses.polar" >}}
-allow(approver: User, "approve", expense: Expense) if
-    approver = expense.{{% exampleGet "submitted_by" %}}.{{% exampleGet "manager" %}}
-    and expense.{{% exampleGet "amount" %}} < 10000;
-{{< /code >}}
-
-In the process of evaluating that rule, the Oso engine would call back into the
-application in order to make determinations that rely on application data, such
-as:
-
-- Which user submitted the expense in question?
-- Who is their manager?
-- Is their manager the user who’s attempting to approve the expense?
-- Does the expense’s `amount` field contain a value less than $100.00?
-
-For more on leveraging application data in an Oso policy, check out
-[Application Types](policies#application-types).
-
 ## Want to talk it through?
 
 If you have any questions, are getting stuck, or just want to talk something
 through, jump into [Slack](https://join-slack.osohq.com/) and an engineer from
 the core team (or one of the hundreds of developers in the growing community)
 will help you out.
 
+## Complete Running Example
```

Comment:
It would be great to put this example into a git repository, so that the user can clone it, and so that we can test it and make sure it works, and continues to work as we make updates. To accomplish this we usually add a repository as a subrepo (check out `docs/examples/quickstart/python`).

Then, you can pull in code from a file: https://github.com/osohq/oso/blob/dcb41c989be7d72248320dfecfebe322d71ec775/docs/content/python/new-roles/getting-started.md

(See the `literalInclude` shortcodes in that file).

Reply:
You could use the existing quickstart repository: https://github.com/osohq/oso-python-quickstart

Maybe put the current code on a branch.

Reply:
We would need to change how it's tested. Right now [this ruby code](https://github.com/osohq/oso/blob/dcb41c989be7d72248320dfecfebe322d71ec775/docs/examples/quickstart/test_quickstarts.rb) tests the quickstarts. It assumes they all have the same structure, which is no longer true after this change.

Probably implementing a few basic tests with pytest or built-in [unittest](https://docs.python.org/3/library/unittest.html#module-unittest)

Reply:
https://github.com/osohq/oso-python-quickstart/pull/1